### PR TITLE
Sampling ratio option

### PR DESCRIPTION
**Pull request by joshuart**
This PR enables users to set an optional sampling_ratio parameter that will determine the ratio of the data that will be profiled. sampling_ratio should be a float greater than 0 and less than or equal to 1. This PR only enables the option to be set. A future PR will implement the actual sampling logic.

### CODE REVIEW

File: dataprofiler/profilers/profiler_options.py [hunk 1]
```diff
@@ -1133,6 +1133,7 @@ def __init__(
         self,
         null_values: dict[str, re.RegexFlag | int] = None,
         column_null_values: dict[int, dict[str, re.RegexFlag | int]] = None,
+        sampling_ratio: float = 0.2,
```
Comment (taylorfturner) [inline dataprofiler/profilers/profiler_options.py:1136]: this is great!

File: dataprofiler/tests/profilers/profiler_options/test_structured_options.py [hunk 2]
```diff
@@ -276,6 +280,22 @@ def test_validate(self):
...
+            "{}.sampling_ratio must be greater than 0.0 and less than or equal to 1.0".format(
+                optpth
+            )
+        ]
+        # Test ratio is not a float
+        option.set({"sampling_ratio": 1})
+        self.assertEqual(expected_error_type, option._validate_helper())
+        # Test ratio is out of range
+        option.set({"sampling_ratio": 2.5})
+        self.assertEqual(expected_error_value, option._validate_helper())
```
Comment (taylorfturner) [inline dataprofiler/tests/profilers/profiler_options/test_structured_options.py:297]: i'd recommend for a slightly more encompassing test to also test the down-side range of the bounds of the option. So you test two things, which are great:

- one right at the top end 
- another past the top end

Should add another test, to fully validate the logic, where `sampling_ratio <= 0.0`

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: dataprofiler/profilers/profiler_options.py [hunk 3]
```diff
@@ -1274,6 +1281,25 @@ def _validate_helper(self, variable_path: str = "StructuredOptions") -> list[str
                 "a re.RegexFlag".format(variable_path)
             )
 
+        if self.sampling_ratio is not None and not isinstance(
+            self.sampling_ratio, float
+        ):
+            errors.append(
+                "{}.sampling_ratio must be either None or an float".format(
```
Comment (JGSweets) [inline dataprofiler/profilers/profiler_options.py:1288]: small fix: `or a float`

Comment (taylorfturner) [reply dataprofiler/profilers/profiler_options.py:1288]: +1

Comment (JGSweets) [reply dataprofiler/profilers/profiler_options.py:1288]: Would it be worth to accept an `int` bc `1` is valid.

Comment (taylorfturner) [reply dataprofiler/profilers/profiler_options.py:1288]: yeah I was thinking that too actually -- I think that's a fair point

File: dataprofiler/tests/profilers/profiler_options/test_profiler_options.py [hunk 4]
```diff
@@ -30,6 +30,8 @@ def test_default_profiler_options(self, *mocks):
                 self.assertFalse(profile.options.properties[column].is_enabled)
             elif column == "null_values" or column == "column_null_values":
                 self.assertIsNone(profile.options.properties[column])
+            elif column == "sampling_ratio":
```
Comment (JGSweets) [inline dataprofiler/tests/profilers/profiler_options/test_profiler_options.py:33]: I think we need to also add tests below in the validate to ensure errors are done properly. Do we also need test to set?

Comment (JGSweets) [reply dataprofiler/tests/profilers/profiler_options/test_profiler_options.py:33]: never mind, this isn't the spot!

File: dataprofiler/tests/profilers/profiler_options/test_structured_options.py (hunk 2 above)
Comment (taylorfturner) [reply dataprofiler/tests/profilers/profiler_options/test_structured_options.py:297]: @joshuart let me know if you have any questions

**Review by taylorfturner — APPROVED** [review]

**Review by JGSweets — APPROVED** [review]

**Pull request merged**